vendor/src/github.com/docker/engine-api/types/types.go

```diff
@@ -274,6 +274,27 @@ type ExecStartCheck struct {
 	Tty bool
 }
 
+// HealthcheckResult stores information about a single run of a healthcheck probe
+type HealthcheckResult struct {
+	Start, End time.Time
+	ExitCode   int    // 0=healthy, 1=unhealthy, 2=starting, -1=error running probe
+	Output     string // Output from last check
+}
+
+// Health states
+const (
+	Starting  = "starting"  // Container is not yet ready
```

Comment:
It seems to me that for the first rendition of this feature, it'd be simpler and offer more flexibility for the future if we only have two states, `healthy` or `unhealthy`, and that state should be based directly on the zero or non-zero exit code of the most recent health check, and leave more complex states (grace period during startup, grace period for intermittent failures, etc) for further discussion once we have a good feeling for exactly what these transitions look like in the real world, and how this feature is going to be used longer-term.